core/chains/evm/logpoller/orm_test.go

```diff
@@ -674,3 +674,133 @@ func BenchmarkLogs(b *testing.B) {
+			LogIndex:       2,
+			BlockHash:      common.HexToHash("0x2"),
+			BlockNumber:    2,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{eventSigB.Bytes(), common.HexToHash("0x0001").Bytes()},
+			EventSig:       eventSigB,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0002"),
+			Data:           []byte(""),
+		},
+	}))
+	require.NoError(t, orm.InsertBlock(common.HexToHash("0x2"), 2, time.Now()))
+
+	logs, err = orm.SelectIndexedLogsWithSigsExcluding(eventSigA, eventSigB, 1, sourceAddr, 0, 3, 0)
+
+	require.NoError(t, err)
+	require.Len(t, logs, 0)
+
+	require.NoError(t, orm.InsertLogs([]logpoller.Log{
+		{
+			EvmChainId:     (*utils.Big)(th.ChainID),
+			LogIndex:       3,
+			BlockHash:      common.HexToHash("0x3"),
+			BlockNumber:    3,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{eventSigA.Bytes(), common.HexToHash("0x0002").Bytes()},
+			EventSig:       eventSigA,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0003"),
+			Data:           []byte("requestID-0002 data"),
+		},
+	}))
+	require.NoError(t, orm.InsertBlock(common.HexToHash("0x3"), 3, time.Now()))
+	logs, err = orm.SelectIndexedLogsWithSigsExcluding(eventSigA, eventSigB, 1, sourceAddr, 0, 3, 0)
+	require.NoError(t, err)
+	require.Len(t, logs, 1)
+	require.Equal(t, logs[0].Data, []byte("requestID-0002 data"))
+
+	require.NoError(t, orm.InsertLogs([]logpoller.Log{
+		{
+			EvmChainId:     (*utils.Big)(th.ChainID),
+			LogIndex:       4,
+			BlockHash:      common.HexToHash("0x3"),
+			BlockNumber:    3,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{eventSigA.Bytes(), common.HexToHash("0x0003").Bytes()},
+			EventSig:       eventSigA,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0004"),
+			Data:           []byte("requestID-0003 data"),
+		},
+	}))
+
+	logs, err = orm.SelectIndexedLogsWithSigsExcluding(eventSigA, eventSigB, 1, sourceAddr, 0, 3, 0)
+	require.NoError(t, err)
+	require.Len(t, logs, 2)
+	require.Equal(t, logs[0].Data, []byte("requestID-0002 data"))
+	require.Equal(t, logs[1].Data, []byte("requestID-0003 data"))
+
+	require.NoError(t, orm.InsertLogs([]logpoller.Log{
+		{
+			EvmChainId:     (*utils.Big)(th.ChainID),
+			LogIndex:       5,
+			BlockHash:      common.HexToHash("0x4"),
+			BlockNumber:    4,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{eventSigB.Bytes(), common.HexToHash("0x0003").Bytes()},
+			EventSig:       eventSigB,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0005"),
+			Data:           []byte("requestID-0003 data"),
+		},
+	}))
+	require.NoError(t, orm.InsertBlock(common.HexToHash("0x4"), 4, time.Now()))
+
+	logs, err = orm.SelectIndexedLogsWithSigsExcluding(eventSigA, eventSigB, 1, sourceAddr, 0, 4, 0)
+	require.NoError(t, err)
+	require.Len(t, logs, 1)
+	require.Equal(t, logs[0].Data, []byte("requestID-0002 data"))
+
+	//No enough confirmations
```

Comment:
typo

Reply:
Fixed in [`e2f04b0` (#8870)](https://github.com/smartcontractkit/chainlink/pull/8870/commits/e2f04b04bf86327a1c1598d65d2cf4d33cfd0629)